components/bl-modal-component/src/index.js

```diff
@@ -0,0 +1,47 @@
+import { useEffect } from 'react';
+
+const { cn } = BackendlessUI.CSSUtils;
+
+export default function ModalComponent({ component, eventHandlers, pods }) {
+  const { display, classList, style, disabled, modalVisibility } = component;
+  const { onClose } = eventHandlers;
+
+  const modalContentPod = pods['modalContent'];
+
+  useEffect(() => {
+    const handleEscClick = e => {
+      if (e.keyCode === 27) {
+        if (onClose) {
+          onClose({ modalVisibility: false });
+        }
+      }
+    };
+
+    document.addEventListener('keydown', handleEscClick);
+
+    return () => document.removeEventListener('keydown', handleEscClick);
+  }, []);
+
+  const handleClick = () => {
+    if (onClose) {
+      onClose({ modalVisibility: false });
+    }
```

Comment:
I'm note sure how it works:

If we don't have a handler - it's not closed by backdrop click
If we have - it's closing.

For me that should works in another way:
If handler exists - we provide used a way to close or not using logic
If handler not exists - we close it every time(or add a property to setup that)

Also there must be an action to close it manually - user must have an option to add a button to pod and close it when he need.